contracts/extensions/BasicIssuanceExtension.sol

```diff
@@ -0,0 +1,228 @@
+/*
+    Copyright 2022 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import { PreciseUnitMath } from "@setprotocol/set-protocol-v2/contracts/lib/PreciseUnitMath.sol";
+import { ISetToken } from "@setprotocol/set-protocol-v2/contracts/interfaces/ISetToken.sol";
+
+import { BaseGlobalExtension } from "../lib/BaseGlobalExtension.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+
+import { IIssuanceModule } from "../interfaces/IIssuanceModule.sol";
+
+/**
+ * @title BasicIssuanceExtension
+ * @author Set Protocol
+ *
+ * Smart contract global extension which provides DelegatedManager owner and 
+ * methodologist the ability to accrue and split issuance and redemption fees at an mutable percentage.
+ */
+contract BasicIssuanceExtension is BaseGlobalExtension {
+    using Address for address;
+    using PreciseUnitMath for uint256;
+    using SafeMath for uint256;
+
+    /* ============ Events ============ */
+
+    event ExtensionInitialized(
+        address _setToken,
+        address _delegatedManager
+    );
+
+    event FeesDistributed(
+        address _setToken,
+        address indexed _ownerFeeRecipient,
+        address indexed _methodologist,
+        uint256 _ownerTake,
+        uint256 _methodologistTake
+    );
+
+    /* ============ State Variables ============ */
+
+    // Instance of BasicIssuanceModule
+    IIssuanceModule public immutable issuanceModule;
+
+    // Mapping from Set Token to DelegatedManager 
+    mapping(ISetToken => IDelegatedManager) public setManagers;
```

Comment:
How do we stop an `attackDelegatedManager` with the same `setToken` set as an `originalDelegatedManager` from overwriting the `originalDelegatedManager` in this data structure? currently `initializeExtension` has no check to prevent this.

Reply:
in `initializeModuleAndExtension` the transaction will revert on module initialization due to lack of permissions. 

in `initializeExtension` we could add 
```
(if _manager[setToken] != address(0)) {
require(setToken.manager() == msg.sender)
}
```

here anyone can write, but only current managers can overwrite. This way the factory initialization process still works but you cannot overwrite with an `attackDelegateManager`

Reply:
Except that a factory migration can be front run here.